vital/types/detected_object_set.h

```diff
@@ -28,202 +28,168 @@ class detected_object_set;
+  /// from high to low. If the optional \p threshold is specified, then all
+  /// detections from the set that are less than the threshold are not in the
+  /// selected set. Note that the selected set may be empty.
+  ///
+  /// The returned vector refers to the actual detections in the set, so if you
+  /// make changes to the selected set, you are also changing the object in the
+  /// set. If you want a clean set of detections, call clone() first.
+  ///
+  /// \param threshold
+  ///   Select all detections with confidence not less than this value. If this
+  ///   parameter is omitted, all detections are selected.
+  ///
+  /// \return List of detections.
+  detected_object_set_sptr select(
+    double threshold = detected_object_type::INVALID_SCORE ) const;
+
+  /// Select detections based on class name.
+  ///
+  /// This method returns a vector of detections that have the specified
+  /// \p class_name. These detections are ordered by descending score for the
+  /// name. Note that the selected set may be empty.
+  ///
+  /// The returned vector refers to the actual detections in the set, so if you
+  /// make changes to the selected set, you are also changing the object in the
+  /// set. If you want a clean set of detections, call clone() first.
+  ///
+  /// \param class_name Class name of detections to be selected.
+  /// \param threshold
+  ///   Select all detections with confidence not less than this value. If this
+  ///   parameter is omitted, all detections are selected.
+  ///
+  /// \return List of detections.
+  detected_object_set_sptr select(
+    const std::string& class_name,
+    double threshold = detected_object_type::INVALID_SCORE ) const;
+
+  /// Scale all detection locations by some scale factor.
+  ///
+  /// This method changes the bounding boxes within all stored detections by
+  /// scaling them by the specified scale factor.
+  ///
+  /// \note
+  ///   Detections in this set can be shared by multiple sets, so scaling the
+  ///   detections in this set will also scale the detection in other sets that
+  ///   share the detections. To avoid this, clone() the set before shifting.
+  ///
+  /// \param scale Scale factor to be applied.
+  ///
+  /// \deprecated
+  [[deprecated]]
+  void scale( double scale_factor );
+
+  /// Shift all detection locations by some translation offset.
+  ///
+  /// This method shifts the bounding boxes within all stored detections by a
+  /// supplied column and row shift.
+  ///
+  /// \note
+  ///   Detections in this set can be shared by multiple sets, so shifting the
+  ///   detections in this set will also shift the detection in other sets that
+  ///   share the detections. To avoid this, clone() the set before shifting.
+  ///
+  /// \param col_shift Column (a.k.a. x, i, width) translation value.
+  /// \param row_shift Row (a.k.a. y, j, height) translation value.
+  ///
+  /// \deprecated
+  [[deprecated]]
+  void shift( double col_shift, double row_shift );
```

Comment:
There are some deprecated methods here. Is now a good time to remote them?

Reply:
Good question? It looks like it is safe to remove these, do you think it makes sense to do so in this PR?

Reply:
I would prefer doing this on a follow-up PR we land after just due to time constraints (VIAME still uses this in 2 places and we're going to need this branch merged in it soon so it will minimize updates required) though don't object to it being done

Reply:
Doing it on another branch is ok with me. Otherwise this looks good. Approved.